app/Http/Controllers/UsersController.php

```diff
@@ -130,20 +186,139 @@ public function show($id)
             new UserTransformer(), [
                 'userAchievements',
                 'allRankHistories',
-                'allScores',
-                'allScoresBest',
-                'allScoresFirst',
                 'allStatistics',
-                'beatmapPlaycounts',
                 'followerCount',
                 'page',
                 'recentActivities',
-                'recentlyReceivedKudosu',
-                'rankedAndApprovedBeatmapsets.beatmaps',
-                'favouriteBeatmapsets.beatmaps',
+                'rankedAndApprovedBeatmapsetCount',
+                'favouriteBeatmapsetCount',
             ]
         );
 
-        return view('users.show', compact('user', 'userArray', 'achievements'));
+        $beatmapsets = [
+            'most_played' => $this->mostPlayedBeatmapsets($user),
+            'ranked_and_approved' => $this->rankedAndApprovedBeatmapsets($user),
+            'favourite' => $this->favouriteBeatmapsets($user),
+        ];
+
+        $scores = [
+            'best' => $this->scoresBest($user, $currentMode),
+            'firsts' => $this->scoresFirsts($user, $currentMode),
+            'recent' => $this->scoresRecent($user, $currentMode),
+        ];
+
+        $kudosu = $this->recentKudosu($user);
+
+        return view('users.show', compact(
+            'user',
+            'userArray',
+            'achievements',
+            'beatmapsets',
+            'scores',
+            'currentMode',
+            'kudosu'
+        ));
+    }
+
+    private function parsePaginationParams($perPage)
+    {
+        $this->user = User::lookup(Request::route('user'), 'id');
+        $this->mode = Request::route('mode') ?? 'osu';
```

Comment:
The others have it as part of query string. Also I think the default should be `$this->user->playmode`.